lib/udp/udp_packet.ml

```diff
@@ -65,7 +65,8 @@ module Marshal = struct
       else Ok ((Cstruct.len payload) + sizeof_udp)
     in
     check_header_len () >>= check_overall_len >>= fun len ->
-    unsafe_fill ~pseudoheader ~payload t udp_buf len;
+    let buf = Cstruct.sub udp_buf 0 Udp_wire.sizeof_udp in
+    unsafe_fill ~pseudoheader ~payload t buf len;
```

Comment:
Thanks for the PR.  Rather than this solution, I'll have `unsafe_fill` limit the size of what it sends on -- that way potential future users of `unsafe_fill` won't be surprised, and the preparation for invocation in `into_cstruct` is a bit clearer.